src/com/redhat/qe/auto/bugzilla/BzChecker.java

```diff
@@ -17,65 +17,58 @@
 import com.redhat.qe.xmlrpc.BaseObject;
 import com.redhat.qe.xmlrpc.Session;
 
+import com.redhat.qe.auto.bugzilla.BugzillaAPI;
 
+import com.google.inject.Inject;
 
 /**
  * Example code to retrieve a bugzilla bug's status, given its ID.  This is for future use with testng, 
  * so that testng can decide whether to execute a test, based on the group annotation (which may contain
  * a bug id), and the status of that bug.  If the status is ON_QA, for example, it can be tested.<br>
- * Example Usage: if (BzChecker.getInstance().getBugState("12345") == BzChecker.bzState.ON_QA) {...
+ * Example Usage: if (BzChecker.getInstance().getBugState("12345") == BugzillaAPI.bzState.ON_QA) {...
  * @author weissj
  *
  */
-public class BzChecker {	
-	
-	public enum bzState { NEW, ASSIGNED, MODIFIED, ON_DEV, ON_QA, VERIFIED, FAILS_QA, RELEASE_PENDING, POST, CLOSED };
-	
+public class BzChecker {
 	protected static Logger log = Logger.getLogger(BzChecker.class.getName());
-	protected static Bug bug;
-	protected static bzState[] defaultFixedBugStates = new bzState[] {
-			BzChecker.bzState.ON_QA,
-			BzChecker.bzState.VERIFIED,
-			BzChecker.bzState.RELEASE_PENDING,
-			BzChecker.bzState.POST,
-			BzChecker.bzState.CLOSED };
-	protected static bzState[] fixedBugStates;
+
+  private final BugzillaAPI bug;
+
+	protected static BugzillaAPI.bzState[] defaultFixedBugStates = new BugzillaAPI.bzState[] {
+			BugzillaAPI.bzState.ON_QA,
+			BugzillaAPI.bzState.VERIFIED,
+			BugzillaAPI.bzState.RELEASE_PENDING,
+			BugzillaAPI.bzState.POST,
+			BugzillaAPI.bzState.CLOSED };
+	protected static BugzillaAPI.bzState[] fixedBugStates;
 	protected static BzChecker instance = null;
-	
-	private BzChecker() {		
-	}
-	
-	private synchronized void init() {
-		bug = new Bug();
+
+  @Inject
+  BzChecker(BugzillaAPI bug){
+      this.bug = bug;
+  }
+
+  private synchronized void init() {
 		try {
-			bug.connectBZ();
-			
+      bug.connectBZ();
+
 			//read in custom "fixed" bug states if any
 			String fixedStates = System.getProperty("bugzilla.fixedBugStates");
 			if (fixedStates != null && fixedStates.length() >0) {
 				fixedBugStates = extractStates(fixedStates);
 			}
 			else fixedBugStates = defaultFixedBugStates;
-			
+
 		}catch(Exception e){
 			throw new RuntimeException("Could not initialize BzChecker." ,e);
 		}
 	}
 
-	public static synchronized BzChecker getInstance(){
-		if (instance == null)	{
-			BzChecker newinst = new BzChecker();
-			newinst.init();
-			instance = newinst;  //if init() fails, don't set the instance
-		}
-		return instance;
+	public BugzillaAPI.bzState getBugState(String bugId) throws Exception{
```

Comment:
I noticed the getInstance function was synchronized, which usually implies some kind of concurrency (the synchronized is like a lock so that only one thread can enter the function at a time).  I'm wondering if that will be necessary for this function too